cpp/src/arrow/scalar_test.cc

```diff
@@ -391,6 +390,8 @@ class TestRealScalar : public ::testing::Test {
     ASSERT_TRUE(list_nan.ApproxEquals(list_other_nan, options));
   }
 
+  void TestListOf() { TestListOf<ListScalar>(list(type_)); }
```

Comment:
Is it me, or is `TestListOf` never called with types other than plain list?

Reply:
And can we add a cast test between map and non-map-list-like somewhere?

Reply:
I added `large_list` here, but note that the cast tests are the ones relevant here.